src/abstractdataframe/selection.jl

```diff
@@ -376,12 +376,60 @@ _transformation_helper(df::AbstractDataFrame, col_idx::Int, (fun,)::Ref{Any}) =
 
 _empty_astable_helper(fun, len) = [fun(NamedTuple()) for _ in 1:len]
 
+@noinline _table_transformation(df_sel::AbstractDataFrame, fun) = fun(Tables.columntable(df_sel))
+
+function _sum_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    return sum(cols)
+end
+
+function _sum_skipmissing_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    sumz = nothing
+    for col in cols
+        try
+            zi = zero(eltype(col))
+            if isnothing(sumz)
+                sumz = zi
+            elseif !ismissing(zi) # zi is missing if eltype is Missing
+                sumz += zi
+            end
+        catch e
+            if e isa MethodError && e.f === Base.zero
+                sumz = nothing
+                break
+            else
+                throw(e)
+            end
+        end
+    end
+    # this will happen if eltype of some columns do not support zero
+    # or all columns have eltype Missing
+    isnothing(sumz) && return nothing
+    init = fill(sumz, length(cols[1]))
+
+    return foldl(cols, init=init) do l, r
+        l .= ifelse.(ismissing.(r), l, l .+ r)
+    end
+end
+
 function _transformation_helper(df::AbstractDataFrame, col_idx::AsTable, (fun,)::Ref{Any})
-    tbl = Tables.columntable(select(df, col_idx.cols, copycols=false))
-    if isempty(tbl) && fun isa ByRow
+    df_sel = select(df, col_idx.cols, copycols=false)
+    if ncol(df_sel) == 0 && fun isa ByRow
         return _empty_astable_helper(fun.fun, nrow(df))
+    elseif typeof(fun) === typeof(Base.sum) || typeof(fun) === ByRow{typeof(Base.sum)}
+        return _sum_fast(map(identity, eachcol(df_sel)))
```

Comment:
```suggestion
        return _sum_fast(collect(eachcol(df_sel)))
```
I guess specialization on column type is intended?

Reply:
I would prefer to use `map` here for the following reason:
```
julia> df = DataFrame(rand(2,3), :auto)
2×3 DataFrame
 Row │ x1        x2        x3
     │ Float64   Float64   Float64
─────┼──────────────────────────────
   1 │ 0.693707  0.583899  0.388987
   2 │ 0.976835  0.576875  0.893499

julia> map(identity, eachcol(df))
3-element Vector{Vector{Float64}}:
 [0.6937071807288211, 0.9768351037375032]
 [0.58389881666453, 0.5768753747948956]
 [0.3889871446622468, 0.8934986529968019]

julia> collect(eachcol(df))
3-element Vector{AbstractVector{T} where T}:
 [0.6937071807288211, 0.9768351037375032]
 [0.58389881666453, 0.5768753747948956]
 [0.3889871446622468, 0.8934986529968019]
```
Why do you think `collect` would be better?

Reply:
I prefer to use `map` because:
```
julia> df = DataFrame(rand(2,3), :auto)
2×3 DataFrame
 Row │ x1        x2        x3
     │ Float64   Float64   Float64
─────┼──────────────────────────────
   1 │ 0.693707  0.583899  0.388987
   2 │ 0.976835  0.576875  0.893499

julia> map(identity, eachcol(df))
3-element Vector{Vector{Float64}}:
 [0.6937071807288211, 0.9768351037375032]
 [0.58389881666453, 0.5768753747948956]
 [0.3889871446622468, 0.8934986529968019]

julia> collect(eachcol(df))
3-element Vector{AbstractVector{T} where T}:
 [0.6937071807288211, 0.9768351037375032]
 [0.58389881666453, 0.5768753747948956]
 [0.3889871446622468, 0.8934986529968019]
```

Reply:
Sorry, I always forget that `collect(x)` isn't always identical to `map(identity, x)`.